### PR TITLE
Add comments to configuration nodes

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by ppalaga** [comment]
@jpkrohling this PR contains many small cosmetic changes with one notable exception of "Add comments to configuration nodes" 1f79ada that I announced earlier.

I also have a followup in `srcdeps-maven` that basically adds a new Maven plugin for auto-generating of `srcdeps.yaml` and `extensions.xml` files.

File: srcdeps-core-config-yaml/src/main/java/org/srcdeps/config/yaml/writer/YamlWriterVisitor.java [hunk 1]
```diff
@@ -63,34 +64,55 @@ public void close() throws IOException {
...
         }
     }
 
+    private void writeComment(List<String> commentLines) throws IOException {
+        if (!commentLines.isEmpty() && depth != 0) {
+            out.write(configuration.getNewLine());
+        }
+        for (String line : commentLines) {
+            indent();
+            out.write('#');
```
Comment (jpkrohling) [inline srcdeps-core-config-yaml/src/main/java/org/srcdeps/config/yaml/writer/YamlWriterVisitor.java:100]: Is it for YAML? If so, can you confirm that comments can be indented? I remember that one of those new formats have problems with the comment char after spaces, but not sure it's YAML.

**Review by jpkrohling — APPROVED** [review]
Looks good to me, it's just not clear to me the actual use case.

File: srcdeps-core-config-yaml/src/main/java/org/srcdeps/config/yaml/writer/YamlWriterVisitor.java (hunk 1 above)
Comment (ppalaga) [reply srcdeps-core-config-yaml/src/main/java/org/srcdeps/config/yaml/writer/YamlWriterVisitor.java:100]: I have no precise knowledge about this. In any case, SnakeYAML can parse indented comments and SnakeYAML seemed to be quite YAML-spec aware as far as I looked into its code.

**Comment by ppalaga** [comment]
@jpkrohling thanks!

**Pull request merged**